dali/aux/optical_flow/optical_flow_adapter_test.cc

```diff
@@ -13,25 +13,53 @@
 // limitations under the License.
 
 #include <gtest/gtest.h>
+#include <dali/aux/optical_flow/optical_flow_stub.h>
+#include <dali/kernels/backend_tags.h>
+#include <dali/util/cuda_utils.h>
 #include <memory>
-#include "dali/aux/optical_flow/optical_flow_stub.h"
 
 namespace dali {
 namespace optical_flow {
 
 using kernels::TensorView;
+using kernels::ComputeCPU;
+using kernels::StorageCPU;
+using kernels::ComputeGPU;
+using kernels::StorageGPU;
 
 static int kTestDataSize = 2;
 
-TEST(OpticalFlowAdapter, StubApi) {
+TEST(OpticalFlowAdapter, StubApiCpuBackend) {
   std::unique_ptr<float> data(new float[kTestDataSize]);
-  TensorView<detail::Backend, uint8_t, 3> tvref, tvin;
-  TensorView<detail::Backend, float, 3> tvout(data.get(), {1, 1, 2});
+  TensorView<StorageCPU, uint8_t, 3> tvref, tvin;
+  TensorView<StorageCPU, float, 3> tvout(data.get(), {1, 1, 2});
   OpticalFlowParams params;
-  std::unique_ptr<OpticalFlowAdapter> of(new OpticalFlowStub(params));
+  std::unique_ptr<OpticalFlowAdapter<ComputeCPU>> of(new OpticalFlowStub<ComputeCPU>(params));
   of->CalcOpticalFlow(tvref, tvin, tvout);
-  EXPECT_FLOAT_EQ(OpticalFlowStub::kStubValue, *tvout(0, 0, 0));
-  EXPECT_FLOAT_EQ(OpticalFlowStub::kStubValue / 2, *tvout(0, 0, 1));
+  EXPECT_FLOAT_EQ(OpticalFlowStub<ComputeCPU>::kStubValue, *tvout(0, 0, 0));
+  EXPECT_FLOAT_EQ(OpticalFlowStub<ComputeCPU>::kStubValue / 2, *tvout(0, 0, 1));
+}
+
+
+TEST(OpticalFlowAdapter, StubApiGpuBackend) {
+  using StubValueType = std::remove_const<decltype(OpticalFlowStub<ComputeGPU>::kStubValue)>::type;
+  StubValueType *tvout_data;
+  CUDA_CALL(cudaMalloc(reinterpret_cast<void **>(&tvout_data),
+                       kTestDataSize * sizeof(StubValueType)));
+
+  TensorView<StorageGPU, uint8_t, 3> tvref, tvin;
+  TensorView<StorageGPU, float, 3> tvout(tvout_data, {1, 1, 2});
+  OpticalFlowParams params;
+  std::unique_ptr<OpticalFlowAdapter<ComputeGPU>> of(new OpticalFlowStub<ComputeGPU>(params));
+  of->CalcOpticalFlow(tvref, tvin, tvout);
+
+  std::unique_ptr<float> host(new float[kTestDataSize]);
```

Comment:
this is leaking. Why not using a vector?

Reply:
Done

Reply:
Not done, it is still leaking

Reply:
Now done :)